### ISSUE TITLE
ChoiceGroup has speacial treatment of aria-labeledby, and should include it as a prop

### ISSUE THREAD

**Issue by BrendanMcK** [OfficeDev/office-ui-fabric-react#6095]
### Bug Report 
 
 
- __Package version(s)__: 6.49.2
- __Browser and OS versions__: any 

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? No
 
Requested priority: Normal
 
Products/sites affected: unknown
 
#### Describe the issue:

We're using the axe-core a11y validation library to do accessibility validation on our component.
 
ChoiceGroup works great if you use its own label - axe-core has a test that ensures all radio buttons are in a group that has a label. Specifically:
- all radiobutton items are within a `role=radiogroup`,
- that radiogroup has an `aria-labelledby` with a valid ID that refers to an element containing text (ie the actual label)
![image](https://user-images.githubusercontent.com/1754373/44682926-982f7d00-a9f9-11e8-9263-a4532659bdee.png)

...so everything validates nicely. Example at: https://codepen.io/brendanmckeon/pen/qMZzqP

However, if you want to use your own custom label - perhaps you have some other component earlier in the DOM that's acting as a label here - then you end up with a `role="radiogroup"` *without* an aria-label, so the axe-core validation fails.
![image](https://user-images.githubusercontent.com/1754373/44683167-55ba7000-a9fa-11e8-9600-15690812971d.png)

Example at: https://codepen.io/brendanmckeon/pen/eLZwRB

One possible fix here would be for ChoiceGroup to take an optional prop where where the id of the custom label can be passed in, and will use that as the `aria-labelledby` value on the `radiogroup` element.

**Comment by micahgodbolt** [comment]
we just made a change in #6035, so we should be sure to take that into account (i.e. don't undo the fix that it applied.)

**Comment by micahgodbolt** [comment]
you can pass in a custom label's ID via aria-describedby

https://codepen.io/micahgodbolt/pen/LJRVbY

**Comment by BrendanMcK** [comment]
Nice - you mean aria-labelledby, though?

The codepen seems to work nicely.

Is this documented? I'm not seeing aria-labelledby listed in the [docs for ChoiceGroup](https://developer.microsoft.com/en-us/fabric#/components/choicegroup)

**Comment by BrendanMcK** [comment]
Got it - I was wondering about the plumbing here - whether it was just getting passed through (but still ended up on the right element, not on the root div of that component) vs was getting special handling.

Also the fact that this was using the raw `aria-labelledby` name vs the usual Fabric camelCase `ariaLabelledby` prop made it look somewhat unusual.

So I'm wondering if the right thing to do here is to add ariaLabelledby as the prop to ChoiceGroup, and have the internals check for ariaLabelledby, falling back to aria-labelledby (eh, is any code *outside* of Fabric using that?)

--

Separately, while I'm here - know anything about the history of the `role=application` in the ChoiceGroup? Those tend to be a major a11y code smell. There's little to no justification in the code as to exactly *why* it's needed.

**Comment by BrendanMcK** [comment]
> Are you intested in picking up this work?

Sounds simple enough - ya, I'll see if I can get around to it in the next few days or so.

--

Re the role=application - if time permits, I'll try removing it and check that it doesn't break JAWS horribly. role=application is really its own special thing apart from the rest of the other aria roles, and is far too often used where its not appropriate. Any legit use really needs to be well commented in the code outlining why it needs to be there, and under what conditions it can be removed in future (eg. JAWS fixes issue XYZ.)
